File: mySolutions/3. Longest Substring Without Repeating Characters/3. Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        String maxStr = "";
        
        for (int i = 0; i < s.length(); i++) {
            String str = "";
            
            for (int j = i; j < s.length(); j++) {
                char nc = s.charAt(j);
                if (i != j && str.indexOf(nc) != -1) {
                    break;
                }
                
                str = str + s.charAt(j);
                
                if (str.length() > maxStr.length()) {
                    maxStr = str + "";
                }
            }
        }
        System.out.print(maxStr);
        return maxStr.length();
    }
}

File: mySolutions/3. Longest Substring Without Repeating Characters/3. Longest Substring Without Repeating Characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        maxLength = 0;
        windowStart = 0;
        charMap = {};
        
        for windowEnd in range(len(s)):
            charMap[s[windowEnd]] = charMap.get(s[windowEnd], 0) + 1;
            
            while charMap[s[windowEnd]] > 1:
                charMap[s[windowStart]] = charMap[s[windowStart]] - 1;
                windowStart += 1;
            
            maxLength = max(windowEnd - windowStart + 1, maxLength);
            
        return maxLength;